File: physioghar_therapist/physioghar_backend/app/schedule_model.py
from sqlalchemy import Column, Integer, String

from app.database import Base


class Schedule(Base):
    __tablename__ = "schedules"

    id = Column(Integer, primary_key=True, index=True)
    date = Column(String, nullable=False)
    time = Column(String, nullable=False)
    status = Column(String, nullable=False)
    patient_id = Column(Integer, nullable=True)
    patient_name = Column(String, nullable=True)
    notes = Column(String, nullable=True)

File: physioghar_therapist/lib/models/schedule_model.dart
class ScheduleModel {
  final int id;
  final String date;
  final String time;
  final String status;
  final int? patientId;
  final String? patientName;
  final String? notes;

  ScheduleModel({
    required this.id,
    required this.date,
    required this.time,
    required this.status,
    this.patientId,
    this.patientName,
    this.notes,
  });

  factory ScheduleModel.fromJson(Map<String, dynamic> json) {
    return ScheduleModel(
      id: json['id'] ?? 0,
      date: json['date'] ?? '',
      time: json['time'] ?? '',
      status: json['status'] ?? '',
      patientId: json['patient_id'],
      patientName: json['patient_name'],
      notes: json['notes'],
    );
  }

  ScheduleModel copyWith({
    int? id,
    String? date,
    String? time,
    String? status,
    int? patientId,
    String? patientName,
    String? notes,
  }) {
    return ScheduleModel(
      id: id ?? this.id,
      date: date ?? this.date,
      time: time ?? this.time,
      status: status ?? this.status,
      patientId: patientId ?? this.patientId,
      patientName: patientName ?? this.patientName,
      notes: notes ?? this.notes,
    );
  }
}
